### DISCUSSION TITLE
Can I self host hubs using my own hardware?

### DISCUSSION THREAD

**Discussion by JeffreyConnected** [mozilla/hubs#2699] [Hubs Cloud Questions] [answered]
Has anyone ever hosted Hubs from a home server to avoid AWS cost? Is this even possible?

**Comment by robinkwilson** [comment, marked as answer]
The client and server code for Hubs is open source, so it's possible to do in theory but it's not a path that we support for organizations since configurations for individual development and production environments can vary significantly.

If you're experimenting, [hubs.mozilla.com](https://hubs.mozilla.com/) is completely free to use!

If you are interested in Hubs Cloud and specifically worried about costs, we have docs about minimizing aws costs: [Minimizing costs user story](https://hubs.mozilla.com/docs/hubs-cloud-aws-costs.html#minimize-your-costs---a-user-story) and monitoring costs via aws's [Billing and Cost Management Dashboard](https://console.aws.amazon.com/billing/home?#/). The Hubs Cloud DigitalOcean (alpha) offering is also cost effective.

**Reply by netpro2k** [reply]
Yep as Robin said, this should be possible to do but is going to be a fair amount of work. 

I think the happy path would probably be using the existing habitat packages and either running them with a local habitat supervisor, or using them to generate docker images for each of the services. The tricky part is going to come in configuring the hundreds of configuration options that exist across the various services and need to work together. We can probably help answer high level questions, but you would be treading in uncharted waters, so the nitty gritty would probably be up to you to figure out.

This is maybe something we would like to tackle ourselves at some point, but its not currently on the direct raodmap

P.S. With some tinkering I was able to take our standard digitalocean offering on a $5/month instance and install postgres locally on the node, getting rid of the $15/month hosted database. It probably can only support a few users, and the database has no redundancy or backup, but this is perfect for my usecase of a silly server I just want to be able to use with a few friends.

**Reply by Sliferblade** [reply]
Any tips on how to connect the local postgres database? or even an external database for that matter? I have only managed to get the server running with the hosted database.

**Comment by misslivirose** [comment]
Some additional discussion about Docker/Habitat as a hosting option: https://github.com/mozilla/hubs/issues/416

**Reply by poVoq** [reply]
Really a pity that there isn't an official way to install via Docker Compose.

Maybe this could be updated to work with the current state: https://github.com/yukimochi-mixedreality/mozilla-hubs-docker

**Comment by albirrkarim** [comment]
I think my [article about hosting hubs on VPS / self-hosted](https://github.com/albirrkarim/mozilla-hubs-installation-detailed/blob/main/VPS_FOR_HUBS.md) can help you

**Reply by elvis0920** [reply]
hey, very detailed article regarding self hosting. just a question regarding the size of Spoke Scene. Is there anyway to remove the 128mb limit when self hosting?

**Reply by albirrkarim** [reply]
Make sure u read the other md file

https://github.com/albirrkarim/mozilla-hubs-installation-detailed/blob/main/PROBLEM_SOLVED.md#--upload-assets-too-large-on-spoke-production
